alibi/explainers/backends/tensorflow/cfrl_base.py

```diff
@@ -479,45 +481,93 @@ def update_actor_critic(ae: keras.Model,
     return losses
 
 
-def to_numpy(x: Optional[Union[List, np.ndarray, tf.Tensor]]) -> Optional[Union[List, np.ndarray]]:
+def to_numpy(X: Optional[Union[List, np.ndarray, tf.Tensor]]) -> Optional[Union[List, np.ndarray]]:
     """
     Converts given tensor to numpy array.
 
     Parameters
     ----------
-    x
+    X
         Input tensor to be converted to numpy array.
 
     Returns
     -------
     Numpy representation of the input tensor.
     """
-    if x is not None:
-        if isinstance(x, np.ndarray):
-            return x
+    if X is not None:
+        if isinstance(X, np.ndarray):
+            return X
 
-        if isinstance(x, tf.Tensor):
-            return x.numpy()
+        if isinstance(X, tf.Tensor):
+            return X.numpy()
 
-        if isinstance(x, list):
-            return [to_numpy(e) for e in x]
+        if isinstance(X, list):
+            return [to_numpy(e) for e in X]
 
-        return np.array(x)
+        return np.array(X)
     return None
 
 
-def to_tensor(x: Union[np.ndarray, tf.Tensor], **kwargs) -> Optional[tf.Tensor]:
+def to_tensor(X: Union[np.ndarray, tf.Tensor], **kwargs) -> Optional[tf.Tensor]:
     """
     Converts tensor to tf.Tensor
 
     Returns
     -------
     tf.Tensor conversion.
     """
-    if x is not None:
-        if isinstance(x, tf.Tensor):
-            return x
+    if X is not None:
+        if isinstance(X, tf.Tensor):
+            return X
 
-        return tf.constant(x)
+        return tf.constant(X)
 
     return None
+
+
+def save_model(path: Union[str, os.PathLike], model: keras.layers.Layer) -> None:
+    """
+    Saves a model and its optimizer.
+
+    Parameters
+    ----------
+    path
+        Path to the saving location.
+    model
+        Model to be saved.
+    """
+    model.save(path, save_format="tf")
+
+
+def load_model(path: Union[str, os.PathLike]) -> keras.Model:
+    """
+    Loads a model and its optimizer.
+
+    Parameters
+    ----------
+    path
+        Path to the loading location.
+
+    Returns
+    -------
+    Loaded model.
+    """
+    return keras.models.load_model(path, compile=False)
+
+
+def set_seed(seed: int = 13):
+    """
+    Sets a seed to ensure reproducibility. Does NOT ensure reproducibility.
+
+    Parameters
+    ----------
+    seed
+        seed to be set
+    """
+    # others
+    os.environ['PYTHONHASHSEED'] = str(seed)
+    random.seed(seed)
+    np.random.seed(seed)
+
+    # tf random
+    tf.random.set_seed(seed)
```

Comment:
Same comment as for `pytorch`, + `to_numpy`, `to_numpy` functions can also be factored out into a generic `tensorflow` utility module.

That does raise the question how to manage it in the public classes as `self.backend.to_numpy` wouldn't work if factored out, unless we do the same trick and import it (`from alibi.explainers.backends.tflow.utils import to_numpy, to_tensor`)?

Reply:
I would say so.